app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java

```diff
@@ -96,11 +97,7 @@ public void onResume() {
             gridView.setVisibility(GONE);
             if (gridAdapter != null) {
                 gridAdapter.clear();
-                try {
-                    ((BookmarksActivity) getContext()).viewPagerNotifyDataSetChanged();
-                }catch (Exception e){
-                    e.printStackTrace();
-                }
+                ((CategoryImagesCallback) getContext()).viewPagerNotifyDataSetChanged();
```

Comment:
Instead of  casting the context to CategoryImagesCallback, should we consider actually passing the callback to the fragment ?

Reply:
That's correct, my mistake.